agent/command/git.go

```diff
@@ -236,6 +242,12 @@ func (opts cloneOpts) buildHTTPCloneCommand(forApp bool) ([]string, error) {
 	if opts.branch != "" {
 		clone = fmt.Sprintf("%s --branch '%s'", clone, opts.branch)
 	}
+	if !opts.cloneAllBranches {
+		clone = fmt.Sprintf("%s --single-branch", clone)
+	}
+	if !opts.fullTreeClone {
+		clone = fmt.Sprintf("%s --filter=tree:0", clone)
```

Comment:
I'm asking runtime env. - but as a worst case scenario, the fallback should just run instead without this option so it should be backwards compatible in that regard

Reply:
It looks like some older (deprecated) distros run  2.13.2 where we would need 2.27 to use --filter (but --single is from like 1.7).  In those cases, it throws an error since it's an unknown option which would bubble up to retrying with the fallback.